### PR TITLE
feat: Remove unit test generation

### LINKED ISSUE #519: Consider removing unit test generation
This should probably only happen after we've got integration tests working with show case, but our unit test generation is frustrating:

- Generating test projects means keeping more dependencies up to date
- Adding a generator feature often means modifying test generation too
- These tests don't actually add value, or at least that value hasn't been noticeable
- Running generated tests of generated code takes significant CI time for no tangible benefit

### PR DESCRIPTION
**Pull request by jskeet**
The unit tests didn't provide value, but *did* take time in terms of dependency updates and writing the code to generate unit tests for new features.

Fixes #519

(Regarding the commit title: it's at least related to a feature... not sure what the conventional commit would be otherwise.)

### CODE REVIEW

**Comment by jskeet** [comment]
We definitely don't want to merge this just yet, but I thought it worth getting ready...

**Review by amanda-tarafa — APPROVED** [review]
LGTM!

**Comment by jskeet** [comment]
Rebased and up-to-date. I've managed to use this (with a few changes to the generation scripts) to successfully generate all APIs in google-cloud-dotnet.

File: Google.Api.Generator/Generation/CsProjGenerator.cs [hunk 1]
```diff
@@ -29,10 +29,6 @@ internal static class CsProjGenerator
         private const string LocationVersion = "[2.0.0, 3.0.0)";
         private const string ReferenceAssembliesVersion = "1.0.2";
         private const string SystemLinqAsyncVersion = "6.0.1";
-        private const string TestSdkVersion = "17.3.1";
```
Comment (amanda-tarafa) [inline Google.Api.Generator/Generation/CsProjGenerator.cs:32]: Yay!

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Pull request merged**